fog/sql_recovery_db/src/lib.rs

```diff
@@ -1005,21 +1003,21 @@ impl SqlRecoveryDb {
             use schema::ingested_blocks::dsl;
             dsl::ingested_blocks
                 .filter(dsl::ingress_public_key.eq(key_bytes))
-                .filter(dsl::block_number.ge(block_index as i64))
-                .limit(block_count as i64)
+                .filter(dsl::block_number.ge(block_range.start_block as i64))
+                .limit(block_range.len() as i64)
```

Comment:
:wrench: might consider storing `block_range.len()` in a local variable and re-using the local variable.